packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_top_up/transform_into_action/v0/mod.rs

```diff
@@ -0,0 +1,181 @@
+use crate::error::Error;
+use crate::platform_types::platform::PlatformRef;
+use crate::rpc::core::CoreRPCLike;
+use dpp::asset_lock::reduced_asset_lock_value::{AssetLockValue, AssetLockValueGettersV0};
+use dpp::balances::credits::CREDITS_PER_DUFF;
+
+use dpp::consensus::signature::{BasicECDSAError, SignatureError};
+use dpp::consensus::state::identity::invalid_asset_lock_proof_value::InvalidAssetLockProofValueError;
+use dpp::dashcore::signer::double_sha;
+use dpp::dashcore::{signer, ScriptBuf};
+use dpp::identity::state_transition::AssetLockProved;
+use dpp::identity::KeyType;
+
+use dpp::prelude::ConsensusValidationResult;
+
+use dpp::state_transition::identity_topup_transition::IdentityTopUpTransition;
+use dpp::state_transition::StateTransitionLike;
+
+use dpp::version::PlatformVersion;
+use drive::state_transition_action::identity::identity_topup::IdentityTopUpTransitionAction;
+use drive::state_transition_action::StateTransitionAction;
+
+use crate::error::execution::ExecutionError;
+use drive::grovedb::TransactionArg;
+
+use crate::execution::types::execution_operation::ValidationOperation;
+use crate::execution::types::execution_operation::signature_verification_operation::SignatureVerificationOperation;
+use crate::execution::types::state_transition_execution_context::{StateTransitionExecutionContext, StateTransitionExecutionContextMethodsV0};
+use crate::execution::validation::state_transition::common::asset_lock::proof::validate::AssetLockProofValidation;
+use crate::execution::validation::state_transition::common::asset_lock::transaction::fetch_asset_lock_transaction_output_sync::fetch_asset_lock_transaction_output_sync;
+use crate::execution::validation::state_transition::ValidationMode;
+
+pub(in crate::execution::validation::state_transition::state_transitions::identity_top_up) trait IdentityTopUpStateTransitionStateValidationV0
+{
+    fn transform_into_action_v0<C: CoreRPCLike>(
+        &self,
+        platform: &PlatformRef<C>,
+        signable_bytes: Vec<u8>,
+        validation_mode: ValidationMode,
+        execution_context: &mut StateTransitionExecutionContext,
+        transaction: TransactionArg,
+        platform_version: &PlatformVersion,
+    ) -> Result<ConsensusValidationResult<StateTransitionAction>, Error>;
+}
+
+impl IdentityTopUpStateTransitionStateValidationV0 for IdentityTopUpTransition {
+    fn transform_into_action_v0<C: CoreRPCLike>(
```

Comment:
The same comments what we have for transform identity create into action

Reply:
Done